src/vs/editor/contrib/suggest/browser/suggestController.ts

```diff
@@ -393,6 +390,9 @@ export class SuggestController implements IEditorContribution {
 			overtypingCapturer: this._overtypingCapturer.value
 		});
 
+		// cancel -> stops all listening and closes widget
+		this.model.cancel();
```

Comment:
The cancelling must happen before calling `insert`. It needs a different change which freezes the overtyping capturer before that.

Reply:
Reverted.
Changed logic of emptying captured selections and added locking while suggestions widget is active.
This should also fix some non-critical cases when it returned previously saved selection when there were no selection at widget trigger time.